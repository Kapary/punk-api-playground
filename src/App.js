import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { connect } from 'react-redux';
import { incrementCounter, decrementCounter } from './actions/buttonActions';

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      beers: [],
      isLoading: false,
    };
  }

  componentDidMount() {
    this.setState({ isLoading: true });

    window.setTimeout(() => {
      fetch('https://api.punkapi.com/v2/beers', {mode: 'cors'})
        .then(response => response.json())
        .catch(error => { console.log('error', error); }  )
        .then( data => { this.setState({'beers': data, isLoading: false }); })
    }, 3000);
  }

  renderList = (elements) => {
    if (!elements.length) {
      return null;
    }

    return elements.map(element => {
      return <div key={element.id}>{element.name}</div>
    });
  }

  renderLoader = () => <div>LOADING...</div>;
  
  render() {
    const { beers, isLoading } = this.state;
    const { pageTitle, onButtonClick, currentCount, rootProp, incrementCounter, decrementCounter } = this.props;

    console.log(currentCount)
    console.log(this.props)

    return (
      <div className="App" rootProp={rootProp}>
        <h1>{pageTitle}</h1>
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <h2>{currentCount}</h2>

          <button id="testing-button" onClick={incrementCounter}>increment me!</button>
          <button id="testing-button" onClick={decrementCounter}>decrement me!</button>
          <button id="testing-button" onClick={onButtonClick}>Click me!</button>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
        { isLoading ? this.renderLoader() : this.renderList(beers)}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  currentCount: state.buttonReducer.currentCount
})

const mapDispatchToProps = dispatch => ({
  incrementCounter: () => dispatch(incrementCounter()),
  decrementCounter: () => dispatch(decrementCounter())
})

export default connect(mapStateToProps, mapDispatchToProps)(App);
