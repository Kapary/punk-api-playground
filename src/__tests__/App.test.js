/* import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
}); */


import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App';

it('renders without crashing', () => {
  shallow(<App />);
});

it('checks the equality', () => {
  expect(1 + 2).toEqual(3);
})

it('should render "New App" as page title ', () => {
  const wrapper = shallow(<App pageTitle="New App"/>);

  // We should use instance when checking props that are not used at the top component level
  expect(wrapper.instance().props.pageTitle).toEqual('New App');
});


it('should check the euqality', () => {
  expect(10+3).toEqual(13);
});

// This style will only work for the props used at the root level like "small" or sth alike
it('should render "New App" as page title ', () => {
  const wrapper = shallow(<App rootProp="WTF"/>);
  expect(wrapper).toMatchSnapshot();
  //expect(wrapper.prop('rootProp')).toEqual('WTF');
});






