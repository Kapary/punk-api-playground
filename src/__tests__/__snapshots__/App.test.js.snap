// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render "New App" as page title  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    rootProp="WTF"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1 />,
        <header
          className="App-header"
        >
          <img
            alt="logo"
            className="App-logo"
            src="logo.svg"
          />
          <p>
            Edit 
            <code>
              src/App.js
            </code>
             and save to reload.
          </p>
          <button
            id="testing-button"
          >
            Click me!
          </button>
          <a
            className="App-link"
            href="https://reactjs.org"
            rel="noopener noreferrer"
            target="_blank"
          >
            Learn React
          </a>
        </header>,
        <div>
          LOADING...
        </div>,
      ],
      "className": "App",
      "rootProp": "WTF",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />,
            <p>
              Edit 
              <code>
                src/App.js
              </code>
               and save to reload.
            </p>,
            <button
              id="testing-button"
            >
              Click me!
            </button>,
            <a
              className="App-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>,
          ],
          "className": "App-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "App-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Edit ",
                <code>
                  src/App.js
                </code>,
                " and save to reload.",
              ],
            },
            "ref": null,
            "rendered": Array [
              "Edit ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "src/App.js",
                },
                "ref": null,
                "rendered": "src/App.js",
                "type": "code",
              },
              " and save to reload.",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Click me!",
              "id": "testing-button",
              "onClick": undefined,
            },
            "ref": null,
            "rendered": "Click me!",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Learn React",
              "className": "App-link",
              "href": "https://reactjs.org",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Learn React",
            "type": "a",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "LOADING...",
        },
        "ref": null,
        "rendered": "LOADING...",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1 />,
          <header
            className="App-header"
          >
            <img
              alt="logo"
              className="App-logo"
              src="logo.svg"
            />
            <p>
              Edit 
              <code>
                src/App.js
              </code>
               and save to reload.
            </p>
            <button
              id="testing-button"
            >
              Click me!
            </button>
            <a
              className="App-link"
              href="https://reactjs.org"
              rel="noopener noreferrer"
              target="_blank"
            >
              Learn React
            </a>
          </header>,
          <div>
            LOADING...
          </div>,
        ],
        "className": "App",
        "rootProp": "WTF",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="logo"
                className="App-logo"
                src="logo.svg"
              />,
              <p>
                Edit 
                <code>
                  src/App.js
                </code>
                 and save to reload.
              </p>,
              <button
                id="testing-button"
              >
                Click me!
              </button>,
              <a
                className="App-link"
                href="https://reactjs.org"
                rel="noopener noreferrer"
                target="_blank"
              >
                Learn React
              </a>,
            ],
            "className": "App-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "App-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Edit ",
                  <code>
                    src/App.js
                  </code>,
                  " and save to reload.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Edit ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "src/App.js",
                  },
                  "ref": null,
                  "rendered": "src/App.js",
                  "type": "code",
                },
                " and save to reload.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Click me!",
                "id": "testing-button",
                "onClick": undefined,
              },
              "ref": null,
              "rendered": "Click me!",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Learn React",
                "className": "App-link",
                "href": "https://reactjs.org",
                "rel": "noopener noreferrer",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Learn React",
              "type": "a",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "LOADING...",
          },
          "ref": null,
          "rendered": "LOADING...",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
